src/browser/modules/Sidebar/relate-scripts.js

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright (c) 2002-2020 "Neo4j,"
+ * Neo4j Sweden AB [http://neo4j.com]
+ * This file is part of Neo4j.
+ * Neo4j is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+import React, { useEffect, useState, useRef } from 'react'
+import { withBus } from 'react-suber'
+import { connect } from 'react-redux'
+import MyScripts from '@relate-by-ui/saved-scripts'
+import { useQuery, gql, useMutation, NetworkStatus } from '@apollo/client'
+import path from 'path'
+
+import * as editor from 'shared/modules/editor/editorDuck'
+import { executeCommand } from 'shared/modules/commands/commandsDuck'
+import { SLASH } from 'shared/services/export-favorites'
+import {
+  getProjectFilesQueryVars,
+  removeProjectFileMutationVars,
+  mapRelateFavorites
+} from './relate-scripts.utils'
+
+const GET_PROJECT_FILES = gql`
+  query GetProject($projectId: String!, $filterValue: String!) {
+    getProject(name: $projectId) {
+      files(
+        filters: [{ field: "extension", type: EQUALS, value: $filterValue }]
+      ) {
+        name
+        directory
+        extension
+        downloadToken
+      }
+    }
+  }
+`
+const DELETE_PROJECT_FILE = gql`
+  mutation RemoveFile($projectId: String!, $filePath: String!) {
+    removeProjectFile(name: $projectId, filePath: $filePath) {
+      name
+      directory
+    }
+  }
+`
+
+const MyScriptsComponent = props => {
+  // @todo: handling loading and error?? Pass to MyScripts??
+  const { loading, error, data, refetch } = useQuery(GET_PROJECT_FILES, {
```

Comment:
error or loading handling suggestions welcome here

Reply:
What can cause errors? Does it only error if the connection is broken? Just thinking how serious the errors are

Reply:
network error like when connection is broken I suppose. Or errors from the graphQL endpoint

Reply:
Hm.. perhaps it's worth having an errorbanner like db not available but for network? For errors from the end point I'm not sure.. We want them in sentry eventually at least

Reply:
yeah the banner makes sense. I will look into that, I didnt look in to the error banners